src/components/form/Admin.jsx

```diff
@@ -18,18 +19,32 @@ const Admin = () => {
     form: "admins",
   });
 
-  const handleSubmit = (setLoading, setState) => {
-    api({
-      method: "POST",
-      url: "/api/dashboard/admins",
-      body: admin,
-    })
-      .then(() => toaster(`Submitted successfully!`, "success"))
-      .catch(() => toaster(`Internal Server Error`, "error"))
-      .finally(() => {
-        setLoading(false);
-        setState(2);
+  const handleSubmit = async (setLoading, setState) => {
+    setLoading(true);
```

Comment:
lines 23 - 32 should be abstracted into a separate function inside utils, since this is a common way to handle the forms parsing we dont need to repeat ourselves so many times